Cannot bump typescript to 5.6 due to botbuilder `INodeBuffer` error

## Versions
Botbuilder-js: 4.23.x
Node: 18.20.4
Browser: N/A
OS: Mac 14.6.x

## Describe the bug
When bumping Typescript to 5.6.x in the Teams AI library, typescript errors caught in the [latest TS update](https://devblogs.microsoft.com/typescript/announcing-typescript-5-6) causes the build to fail due to incorrect extend of `Uint8Array`. Error message: 

```sh
 | ../../node_modules/botframework-streaming/lib/interfaces/INodeBuffer.d.ts(15,18): error TS2430: Interface 'INodeBuffer' incorrectly extends interface 'Uint8Array'.
 |   The types returned by 'slice(...)' are incompatible between these types.
 |     Type 'this' is not assignable to type 'Uint8Array'.
 |       Type 'INodeBuffer' is not assignable to type 'Uint8Array'.
 |         The types returned by 'entries()' are incompatible between these types.
 |           Type 'IterableIterator<[number, number]>' is missing the following properties from type 'ArrayIterator<[number, number]>': map, filter, take, drop, and 9 more.
```

## To Reproduce
Steps to reproduce the behavior:
1. Open any Botbuilder TS sample
2. In `package.json`, upgrade typescript to 5.6.x
3. (If applicable) Remove `node_modules`/ lock files before building the sample
4. Install && build the sample
5. See error in console

--- a/libraries/botframework-streaming/etc/botframework-streaming.api.md
+++ b/libraries/botframework-streaming/etc/botframework-streaming.api.md
@@ -4,6 +4,8 @@
 
 ```ts
 
+/// <reference types="node" />
+
 import { Duplex } from 'stream';
 import { DuplexOptions } from 'stream';
 import { default as WebSocket_2 } from 'ws';

--- a/libraries/botframework-streaming/etc/botframework-streaming.api.md
+++ b/libraries/botframework-streaming/etc/botframework-streaming.api.md
@@ -4,8 +4,6 @@
 
 ```ts
 
-/// <reference types="node" />
-
 import { Duplex } from 'stream';
 import { DuplexOptions } from 'stream';
 import { default as WebSocket_2 } from 'ws';
@@ -44,8 +42,6 @@ export interface INodeBuffer extends Uint8Array {
     // (undocumented)
     copy(targetBuffer: Uint8Array, targetStart?: number, sourceStart?: number, sourceEnd?: number): number;
     // (undocumented)
-    entries(): IterableIterator<[number, number]>;
-    // (undocumented)
     equals(otherBuffer: Uint8Array): boolean;
     // (undocumented)
     fill(value: any, offset?: number, end?: number): this;
@@ -53,8 +49,6 @@ export interface INodeBuffer extends Uint8Array {
     includes(value: string | number | this, byteOffset?: number, encoding?: string): boolean;
     // (undocumented)
     indexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: string): number;
-    // (undocumented)
-    keys(): IterableIterator<number>;
     // (undocumented)
     lastIndexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: string): number;
     // (undocumented)
@@ -108,8 +102,6 @@ export interface INodeBuffer extends Uint8Array {
     };
     // (undocumented)
     toString(encoding?: string, start?: number, end?: number): string;
-    // (undocumented)
-    values(): IterableIterator<number>;
     // (undocumented)
     write(string: string, offset?: number, length?: number, encoding?: string): number;
     // (undocumented)

--- a/libraries/botframework-streaming/src/interfaces/INodeBuffer.ts
+++ b/libraries/botframework-streaming/src/interfaces/INodeBuffer.ts
@@ -96,8 +96,5 @@ export interface INodeBuffer extends Uint8Array {
     fill(value: any, offset?: number, end?: number): this;
     indexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: string): number;
     lastIndexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: string): number;
-    entries(): IterableIterator<[number, number]>;
     includes(value: string | number | this, byteOffset?: number, encoding?: string): boolean;
-    keys(): IterableIterator<number>;
-    values(): IterableIterator<number>;
 }

